src/lib/dummy_preload/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+# Normally, the shim is linked to the openssl or libc preload libs and the shim
+# will be injected when those libs are preloaded. However, if the preloading of
+# both of those libs is disabled, then we inject the dummy lib instead (which is
+# also linked to the shim) in order to ensure that the shim is still loaded into
+# the managed process space as required for Shadow to work properly.
+add_library(shadow_dummy SHARED dummy.c)
```

Comment:
I'm not sure that we want the dummy preload lib. We could instead just always require that the libc preload lib is always preloaded.

Reply:
One reason we might want to disable the libc preload could be if we think the libc call behaves differently to the direct syscall and we want to test shadow with the real libc call. But maybe the solution to this would be to just comment out everything in `syscall_wrappers.c` instead.

Reply:
We decided that it is useful to be able to disable the libc preload lib, but we will keep it as experimental for now. So we'll need a dummy lib to inject the shim in case no other preload is used.